src/frontend/views/games_view.py

```diff
@@ -74,6 +59,34 @@ def get_context_data(self, **kwargs):
         context["sgfplayer_url_name"] = "sgfplayer_rating" # urls.py
       return context
 
+class GamesListByUserView(ListView):
+  template_name = "pages/games_list_by_user.html"
+  paginate_by = 100
+
+  def get_queryset(self):
+    self.run = get_object_or_404(Run, name=self.kwargs["run"])
+    self.user = get_object_or_404(User, username=self.kwargs["user"])
+
+    if self.kwargs["kind"] == "training":
+      games = TrainingGame.objects.filter(submitted_by=self.user,run=self.run).order_by("-created_at")
+    else:
+      games = RatingGame.objects.filter(submitted_by=self.user,run=self.run).order_by("-created_at")
+    return games
```

Comment:
Also not sure weither it is bettr to have this kind logic vs having two views.

```py
class TrainingGamesListByUserView(ListView): 
  template_name = "pages/training_games_list_by_user.html"
  paginate_by = 100

  def get_queryset(self):
    self.run = get_object_or_404(Run, name=self.kwargs["run"])
    self.user = get_object_or_404(User, username=self.kwargs["user"])
    return TrainingGame.objects.filter(submitted_by=self.user,run=self.run).order_by("-created_at")
``` 

I think by putting if we decrease cohesion and we are less clear.